megamek/src/megamek/client/ui/swing/OffBoardTargetOverlay.java

```diff
@@ -0,0 +1,365 @@
+     */
+    private boolean showDirectionalElement(OffBoardDirection direction, Mounted selectedArtilleryWeapon) {
+        for(Entity entity : getCurrentGame().getAllOffboardEnemyEntities(getCurrentPlayer())) {
+            if(entity.isOffBoardObserved(getCurrentPlayer().getTeam()) && 
+                    (entity.getOffBoardDirection() == direction) &&
+                        (targetingPhaseDisplay.ce().isOffBoard() ||
+                        weaponFacingInDirection(selectedArtilleryWeapon, direction))) {
+                return true;
+            }
+        }
+        
+        return false;
+    }
+    
+    /**
+     * Worker function that determines if the given weapon is facing in the correct off-board direction.
+     */
+    private boolean weaponFacingInDirection(Mounted artilleryWeapon, OffBoardDirection direction) {
+        Coords checkCoords = artilleryWeapon.getEntity().getPosition();
+        int translationDistance;
+        
+        // little hack: we project a point 10 hexes off board to the north/south/east/west and declare a hex target with it
+        // then use Compute.isInArc, as that takes into account all the logic including torso/turret twists.
+        switch(direction) {
+        case NORTH:
+            checkCoords = checkCoords.translated(0, checkCoords.getY() + 10);
+            break;
+        case SOUTH:
+            checkCoords = checkCoords.translated(3, getCurrentGame().getBoard().getHeight() - checkCoords.getY() + 10);
+            break;
+        case EAST:
+            translationDistance = ((getCurrentGame().getBoard().getWidth() - checkCoords.getX()) / 2) + 5;
+            checkCoords = checkCoords.translated(1, translationDistance).translated(2, translationDistance);
+            break;
+        case WEST:
+            translationDistance = checkCoords.getX() + 5;
+            checkCoords = checkCoords.translated(4, translationDistance).translated(5, translationDistance);
+            break;
+        default:
+            return false;
+        }
+        
+        Targetable checkTarget = new HexTarget(checkCoords, Targetable.TYPE_HEX_ARTILLERY);
+        
+        return Compute.isInArc(getCurrentGame(), artilleryWeapon.getEntity().getId(), 
+                artilleryWeapon.getEntity().getEquipmentNum(artilleryWeapon), checkTarget);
+    }
+    
+    @Override
+    public boolean isBeingDragged() {
+        return false;
+    }
+
+    @Override
+    public boolean isDragged(Point point, Dimension backSize) {
+        return false;
+    }
+
+    @Override
+    public boolean isHit(Point point, Dimension size) {
+        Point actualPoint = point;
+        actualPoint.x = (int) (point.getX() + clientgui.getBoardView().getDisplayablesRect().getX());
+        actualPoint.y = (int) (point.getY() + clientgui.getBoardView().getDisplayablesRect().getY());
+        
+        for(OffBoardDirection direction : OffBoardDirection.values()) {
+            if(direction != OffBoardDirection.NONE) {
```

Comment:
If `buttons` will never contain `OffBoardDirection.None` you can probably omit this if test.